File: day_2/e.js
const fs = require("fs");

const exercise_1 = (inputs) => {
    let depth = 0;
    let distance = 0;
    for (const row of inputs) {
        const matches = row.match(/(\w+) (\d+)/);
        switch(matches[1]) {
            case "forward":
                distance += Number(matches[2]);
                break;
            case "up":
                depth -= Number(matches[2]);
                break;
            case "down":
                depth += Number(matches[2]);
                break;
            default:
                console.error("ERROR")
        }
    }
    return depth * distance;
}

const exercise_2 = (inputs) => {
    depth = 0;
    distance = 0;
    let aim = 0;
    for (const row of inputs) {
        const matches = row.match(/(\w+) (\d+)/);
        switch(matches[1]) {
            case "forward":
                distance += Number(matches[2]);
                depth += aim * Number(matches[2]);
                break;
            case "up":
                aim -= Number(matches[2]);
                break;
            case "down":
                aim += Number(matches[2]);
                break;
            default:
                console.error("ERROR")
        }
    }
    return depth * distance;
}


const input = fs.readFileSync("input.txt", { encoding: "utf-8" });

const inputs = input.trim().split("\n");

console.log("Exercise 1:", exercise_1(inputs));
console.log("Exercise 2:", exercise_2(inputs));

File: day_2/e.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func exercise_1(inputs []string) int {
	depth, distance := 0, 0

	for _, row := range inputs {
		tokens := strings.Split(row, " ")
		v, err := strconv.Atoi(tokens[1])
		if err != nil {
			panic(err)
		}

		switch tokens[0] {
		case "forward":
			distance += v
		case "down":
			depth += v
		case "up":
			depth -= v
		}
	}

	return depth * distance
}

func exercise_2(inputs []string) int {
	depth, distance, aim := 0, 0, 0

	for _, row := range inputs {
		tokens := strings.Split(row, " ")
		v, err := strconv.Atoi(tokens[1])
		if err != nil {
			panic(err)
		}

		switch tokens[0] {
		case "forward":
			distance += v
			depth += aim * v
		case "down":
			aim += v
		case "up":
			aim -= v
		}
	}

	return depth * distance
}

func main() {
	data, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}

	inputs := strings.Split(strings.Trim(string(data), " \n"), "\n")

	fmt.Println("Exercise 1:", exercise_1(inputs))
	fmt.Println("Exercise 2:", exercise_2(inputs))
}
